src/NetscapeBookmarkParser.php

```diff
@@ -0,0 +1,250 @@
+
+    private $defaultContext = [
+        NetscapeBookmarkDecoder::KEEP_NESTED_TAGS => true,
+        NetscapeBookmarkDecoder::NORMALIZE_DATES  => true,
+        NetscapeBookmarkDecoder::DATE_RANGE       => '30 years',
+    ];
+
+    /**
+     * @var LoggerInterface instance.
+     */
+    private $logger;
+
+    /**
+     * @var NetscapeBookmarkDecoder
+     */
+    private $decoder;
+
+    /**
+     * @var BookmarksDataFormatter
+     */
+    private $formatter;
+
+    /**
+     * @var NetscapeBookmarkEncoder
+     */
+    private $encoder;
+
+    public function __construct(array $defaultContext = [], LoggerInterface $logger = null)
+    {
+        $this->defaultContext = array_merge($this->defaultContext, $defaultContext);
+        $this->logger = $logger ?? new NullLogger();
+
+        $this->decoder = new NetscapeBookmarkDecoder($this->defaultContext, $this->logger);
+        $this->formatter = new BookmarksDataFormatter();
+        $this->encoder = new NetscapeBookmarkEncoder();
+    }
+
+    /**
+     * Parses a Netscape bookmark file
+     *
+     * @param string $filePath Bookmark file to parse
+     *
+     * @return array An associative array containing parsed links
+     */
+    public function parseFile(string $filePath): array
+    {
+        $this->logger->info('Starting to parse ' . $filePath);
+
+        try {
+            return $this->decoder->decode(file_get_contents($filePath));
+        } catch (\Exception $exception) {
+            $this->logger->error($exception);
+        }
+
+        return [];
+    }
+
+    /**
+     * Exports data to Netscape bookmark format file
+     *
+     * @param array  $data
+     * @param string $filePath
+     *
+     * @return string A string containing serialized data
+     */
+    public function export(array $data, string $filePath): string
+    {
+        $this->logger->info('Export to ' . $filePath);
+
+        $data = $this->formatter->format($data);
+
+
+        try {
+            $data = $this->encoder->encode($data);
+            var_dump($data);
+            die();
+        } catch (\Exception $exception) {
+            $this->logger->error($exception);
+        }
+
+        file_put_contents($filePath, $data);
+
+        return $data;
+    }
+
+    /**
+     * Parses a string containing Netscape-formatted bookmarks
+     *
+     * @param string $data
+     *
+     * @return array
+     * @deprecated
+     */
+    public function parseString(string $data): array
+    {
+        @trigger_error(
+            sprintf(
+                'Method "%s()" is deprecated and will be removed in version %s',
+                __FUNCTION__,
+                self::DEPRECATION_VERSION,
+            ),
+            \E_USER_DEPRECATED
+        );
+
+        try {
+            return $this->decoder->decode($data);
+        } catch (\Exception $exception) {
+            $this->logger->error($exception);
+        }
+
+        return [];
+    }
+
+    /**
+     * @deprecated
```

Comment:
I'm not sure it's necessary to keep those, as the next released version will be 4.0.

Reply:
Depends on the release flow... I believe it is good practice to warn user about methods that will break before bumping version. :-/

Reply:
Oh ! And, I assumed the parser will be removed at some point. Future implementation should use Encoder and Decoder directly which makes more sense IMHO.